docs/agent-reference/agent-core/configuration-manager.md

```diff
@@ -0,0 +1,95 @@
+# Device Update Agent Configuration
+
+## APIs
+
+```c
+// A config object handle.
+typedef void* CONFIG_HANDLE
+
+// Instancitate a config object from specified file.
+CONFIG_HANDLE LoadConfig(const char * configFilePath);
+
+// Free all resources allocated for the config object.
+void FreeConfig(CONFIG_HANDLE config);
+
+// Returns value of the specified config name.
+// (Output is a json string).
+const char* GetValue(const char* name);
+
+```
+
+## Configuration File Format
+
+See example `du-config.json` file format below:
+
+```json
+
+"enabledIoTHubTracing" : true,
+"logLevel" : 0,
+
+"aduShellTrustedUsers": ["adu", "do"],
+
+"agents":[
+  {
+      "name":"host-update",
+      "runas":"adu",
+      "connectionInfoSource":"AIS",
+      "connectionData":{
+          "connectionType" : "module",
```

Comment:
>connectionType" : "module", [](start = 11, length = 27)

We don't decide if we're running as a module AIS does. AIS requires that we pre-provision ourselves with it so it can prepare all the other work. 

We would have to have a startup task that creates all of the required (principals for AIS/IotEdge v1.2) for AIS/IotEdge but that would expect that we 

1) Have a startup task to take care of this
2) Don't see many changes or have the startup pre-provisioning of AIS/IotEdge principals run each time DU is restarted.

Reply:
@nihemstr, so, sounds like only "connectionData.moduleName" property is needed here?